test/lsp/test_query.py signature checks compare `tuple[str, ...]` against `tuple`

## Symptom

Running ``pytest test/lsp/test_query.py`` against the current main
(984fa94) fails two signature-drift checks under Python 3.13:

\`\`\`
FAILED test/lsp/test_query.py::test_splittable_vars_at_signature
FAILED test/lsp/test_query.py::test_eliminable_vars_at_signature

AssertionError: eliminable_vars_at return annotation drifted:
  expected <class 'tuple'>, got tuple[str, ...]
  assert tuple[str, ...] == <class 'tuple'>
\`\`\`

Both ``splittable_vars_at`` and ``eliminable_vars_at`` declare
``-> tuple[str, ...]`` in [lsp/query.py](lsp/query.py); the corresponding
tests in [test/lsp/test_query.py:239](test/lsp/test_query.py) and
[test/lsp/test_query.py:263](test/lsp/test_query.py) pass the bare
``tuple`` class as ``expected_return`` to ``_check_sig``.

Under Python 3.13, ``inspect.signature(func).return_annotation``
gives the parameterised alias ``tuple[str, ...]`` (a ``types.GenericAlias``),
and ``tuple[str, ...] == tuple`` is ``False`` -- so the strict ``==``
in ``_check_sig`` rejects it. Every sibling test in the file passes
the matching parameterised type (e.g. ``list[Diagnostic]``,
``Optional[WorkspaceEdit]``); these two stand out as bare ``tuple``.

## Reproducer

\`\`\`sh
git checkout main  # 984fa94
python3.13 -m pytest test/lsp/test_query.py::test_splittable_vars_at_signature test/lsp/test_query.py::test_eliminable_vars_at_signature
\`\`\`

## Where it lives / suggested fix

[test/lsp/test_query.py:239-244](test/lsp/test_query.py:239) and
[test/lsp/test_query.py:263-268](test/lsp/test_query.py:263): change
the ``expected_return`` argument from bare ``tuple`` to
``tuple[str, ...]`` so it matches the function's actual annotation
and the convention every other ``test_*_signature`` test follows.

## Impact

Local-only papercut. CI doesn't run ``pytest`` on ``test/lsp/`` (see
[.github/workflows/test_deduce.yml](.github/workflows/test_deduce.yml)
-- it runs ``ruff``, ``mypy``, and the ``test-deduce.py`` regression
categories), so these failures don't gate any PR. They do surface
for any developer running the lsp test suite locally, where they
muddy the "did my change break something?" signal -- worth fixing
for that reason.

Spotted while wrapping up [#736](https://github.com/jsiek/deduce/pull/736)
(issue #734); the rest of the lsp suite was green there.

--- a/test/lsp/test_query.py
+++ b/test/lsp/test_query.py
@@ -240,7 +240,7 @@ def test_splittable_vars_at_signature():
     _check_sig(
         splittable_vars_at,
         ["path", "content", "pos", "prelude"],
-        tuple,
+        tuple[str, ...],
     )
 
 
@@ -264,7 +264,7 @@ def test_eliminable_vars_at_signature():
     _check_sig(
         eliminable_vars_at,
         ["path", "content", "pos", "prelude"],
-        tuple,
+        tuple[str, ...],
     )
 
 

